aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -153,22 +167,46 @@ module AptosFramework::Stake {
     /// 3. on_new_epoch processes two pending queues and refresh ValidatorInfo from the owner's address.
     struct ValidatorSet has key {
         consensus_scheme: u8,
-        // active validators for the current epoch
+        // active validators for the current epoch.
         active_validators: vector<ValidatorInfo>,
-        // pending validators to leave in next epoch (still active)
+        // pending validators to leave in next epoch (still active).
         pending_inactive: vector<ValidatorInfo>,
-        // pending validators to join in next epoch
+        // pending validators to join in next epoch.
         pending_active: vector<ValidatorInfo>,
     }
 
+    /// Represents the validator set's current epoch voting power and changes.
+    /// This is a separate resource from ValidatorSet to optimize the cost of reading as it'll be read when individual
+    /// validators join/leave/add stake/withdraw stake. ValidatorSet can contain many validators and thus are expensive
+    /// to read.
+    /// This can reduce efficiency gained from parallel execution if there are multiple validator txs that change the
+    /// voting power being executed together. There are, however, no other strategies to tracking the voting power
+    /// changes so we can apply limits appropriately.
+    struct ValidatorSetVotingPower has key {
+        // The total of all validators' voting power in the current epoch.
+        // u128 since the total supply of Aptos coin is u128.
+        total: u128,
+        // We track the withdrawing and joining voting power separately as they have different effect on the network's
+        // security. E.g. withdrawing is direct reduction of voting power but joining is not necessarily direct addition
+        // depending on whether the joining validators are ready to participate in time when the next epoch starts.
+        // Therefore, withdrawing usually affects the network's security while joining can affect the network's liveness
+        // There are also potential different limits on how much voting power can be withdrawing or joining the
+        // validator set.
+        withdrawing: u128,
+        joining: u128,
```

Comment:
should we unify on one word - joining or increase?

Reply:
Updated to be consistently joining.